crates/wasmtime/src/runtime/component/component.rs

```diff
@@ -20,8 +20,36 @@ use wasmtime_runtime::{
 };
 
 /// A compiled WebAssembly Component.
-//
-// FIXME: need to write more docs here.
+///
+/// This structure represents a compiled component that is ready to be
+/// instantiated. This owns a region of virtual memory which contains executable
+/// code compiled from a WebAssembly binary originally. This is the analog of
+/// [`Module`](crate::Module) in the component embedding API.
+///
+/// A [`Component`] can be turned into an
+/// [`Instance`](crate::component::Instance) through a
+/// [`Linker`](crate::component::Linker). [`Component`]s are safe to share
+/// across threads. The compilation model of a component is the same as that of
+/// [a module](crate::Module) which is to say:
+///
+/// * Compilation happens synchronously during [`Component::new`].
+/// * The result of compilation can be saved into storage with
+///   [`Component::serialize`].
+/// * A previously compiled artifact can be parsed with
+///   [`Component::deserialize`].
+/// * No compilation happens at runtime for a component, everything is done
+///   after [`Component::new`] returns.
```

Comment:
I initially read this as (what I assume is) the opposite of what it means. How about:

```suggestion
/// * No compilation happens at runtime for a component — everything is done
///   by the time [`Component::new`] returns.
```